be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1199,6 +1199,34 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
     return Status::OK();
 }
 
+void SegmentIterator::_convert_dict_code_for_predicate_if_necessary() {
+    for (auto predicate : _short_cir_eval_predicate) {
+        _convert_dict_code_for_predicate_if_necessary_impl(predicate);
+    }
+
+    for (auto predicate : _pre_eval_block_predicate) {
+        _convert_dict_code_for_predicate_if_necessary_impl(predicate);
+    }
+
+    std::set<const ColumnPredicate*> predicate_set {};
+    _opts.delete_condition_predicates->get_all_column_predicate(predicate_set);
+    for (const auto column_predicate : predicate_set) {
```

Comment:
Could you please also check if  dev-1.1.2 also have this problem?

Reply:
Yes, dev-1.1.2 has the same problem.